pkgs/tools/nix/nix-monitored/monitored.cc

```diff
@@ -0,0 +1,77 @@
+#include <libgen.h>
+#include <stdlib.h>
+#include <string>
+#include <sys/wait.h>
+#include <unistd.h>
+#include <vector>
+
+void execvp_array(char* args[]) {
+	args[0] = basename(args[0]);
+	execvp(args[0], args);
+	exit(127);
+}
+
+void execvp_vector(std::vector<char*>&& args) {
+	execvp_array(args.data());
+}
+
+int main(int argc, char* argv[]) {
+	std::string const path(std::string(PATH) + ":" + getenv("PATH"));
+	setenv("PATH", path.c_str(), 1);
+	if (!isatty(fileno(stderr)) || argc < 2) {
+		execvp_array(argv);
+	}
+	std::string const verb(argv[1]);
+	if (verb == "run") {
```

Comment:
This matching based on the argument feels very brittle, especially since the new CLI isn't stable, and only a subset of commands can be piped to `nom`. A better approach imo would be to have tighter integration into Nix itself, e.g. by Nix supporting an `output-formatter` option where one can specify `nom` to be used.

Reply:
I agree with the brittleness point; currently this doesn't even handle situations where the verb is in any location other than `argv[1]`; however, at least in my usage, this is already Good Enough™ to be a tangible improvement in UX.